nifi-nar-bundles/nifi-update-attribute-bundle/nifi-update-attribute-processor/src/main/java/org/apache/nifi/processors/attributes/UpdateAttribute.java

```diff
@@ -424,59 +419,104 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         // because is the original flowfile is used for all matching rules. in this
         // case the order of the matching rules is preserved in the list
         final Map<FlowFile, List<Rule>> matchedRules = new HashMap<>();
-        Map<String, String> statefulAttributes = null;
+
+        final Map<String, String> stateInitialAttributes;
+        final Map<String, String> stateWorkingAttributes;
+        StateMap stateMap = null;
 
         matchedRules.clear();
```

Comment:
Lol, yeah that isn't necessary. I'll remove it. It is a remanent of when we removed the "session.get(100)" at the beginning of the onTrigger when run duration wasn't a thing.